zipline/data/history_loader.py

```diff
@@ -380,7 +380,7 @@ def _ensure_sliding_windows(self, assets, dts, field,
         assets = self._asset_finder.retrieve_all(assets)
 
         try:
-            end_ix = self._calendar.get_loc(end)
+            end_ix = self._calendar.searchsorted(end)
```

Comment:
Thanks for spotting this one. I was unaware of that bug/behavior.

On the same track, I do question whether we should pull in all minutes ever from the calendar (mea culpa), in favor of calculating the location in a similar fashion to the minute bar reader's find position of minute.
(The minute index takes 11MB at minimum for 14 years just for the datetime values.)

Does using `getitem` also trigger the creation of the hash table?
If so we should also audit usage of `trading_calendar.previous_minute`.